### PR TITLE
[bleutrade] started adding v3 endpoints, [coinbase] minor tidy up

### PR DESCRIPTION
**Pull request by npomfret**
@kroitor I've started adding the bleutrade v3 api.  The API itself isn't very good (or the docs are incomplete).  Anyway, this PR doesn't really add much.  It only pokes their ledger, but as paging isn't implemented you can only get 100 transactions out.  It's of limited use to anyone but I wanted to get the PR in before I do something silly and lose the code.

### CODE REVIEW

File: js/bleutrade.js [hunk 1]
```diff
@@ -306,6 +328,105 @@ module.exports = class bleutrade extends bittrex {
...
+        }
+        const description = item['Description'];
+        const descParts = description.split (',');
+        const type = this.parseLedgerEntryType (this.safeString (item, 'Type'));
+        let referenceId = undefined;
+        let fee = undefined;
+        for (let i = 0; i < descParts.length; i++) {
+            const part = descParts[i].trim ();
+            if (part.startsWith ('fee')) {
+                const feeCost = this.safeFloat (part.replace ('fee -', '').trim ());
```
Comment (kroitor) [inline js/bleutrade.js]: It ignores positive fees?

File: js/bleutrade.js [hunk 2]
```diff
@@ -306,6 +328,105 @@ module.exports = class bleutrade extends bittrex {
         };
     }
 
+    parseLedgerEntryType (type) {
+        // deposits don't seem to appear in here
+        const types = {
+            'TRADE': 'trade',
+            'WITHDRAW': 'withdrawal',
```
Comment (kroitor) [inline js/bleutrade.js]: Need to change this a bit, for a ledger entry it is `transaction`, not `withdrawal`/`deposit`. The direction indicates whether it's a deposit or a withdrawal, but the type if `transaction, meaning a blockchain thing.

File: js/bleutrade.js [hunk 3]
```diff
@@ -306,6 +328,105 @@ module.exports = class bleutrade extends bittrex {
...
+                };
+            } else if (part.startsWith ('order id') && type === 'trade') {
+                referenceId = descParts[1].replace ('order id', '').trim ();
+            }
+        }
+        const time = this.safeString (item, 'TimeStamp');
+        const timestamp = this.parse8601 (time);
+        const amount = this.safeFloat (item, 'Amount');
+        let direction = undefined;
+        if (amount < 0) {
```
Comment (kroitor) [inline js/bleutrade.js]: We have to `Math.abs (amount)` for now, because it's documented as an absolute number.

File: js/bleutrade.js [hunk 4]
```diff
@@ -306,6 +328,105 @@ module.exports = class bleutrade extends bittrex {
...
+        //     CoinSymbol: 'DOGE',
+        //     CoinName: 'Dogecoin' }
+        //
+        const currencyId = this.safeString (item, 'CoinSymbol');
+        let code = undefined;
+        if (currencyId !== undefined) {
+            code = this.commonCurrencyCode (currencyId.toUpperCase ());
+        }
+        const description = item['Description'];
+        const descParts = description.split (',');
```
Comment (kroitor) [inline js/bleutrade.js]: The withdrawal description seems to be separated by `';'`.

File: js/bleutrade.js (hunk 3 above)
Comment (kroitor) [reply js/bleutrade.js]: Ah, now I see this below.

File: js/bleutrade.js (hunk 1 above)
Comment (npomfret) [reply js/bleutrade.js]: Not ignoring positive fees.  Just making all fee's positive.

File: js/bleutrade.js (hunk 2 above)
Comment (npomfret) [reply js/bleutrade.js]: Got it. Thanks.

File: js/bleutrade.js (hunk 1 above)
Comment (kroitor) [reply js/bleutrade.js]: What if it starts with `fee ` but not `fee -`?

File: js/bleutrade.js (hunk 4 above)
Comment (npomfret) [reply js/bleutrade.js]: Thanks

File: js/bleutrade.js (hunk 1 above)
Comment (npomfret) [reply js/bleutrade.js]: Have fixed

Comment (kroitor) [reply js/bleutrade.js]: @npomfret thank you!

**Comment by kroitor** [comment]
@npomfret hope to merge it within a day.

File: js/bleutrade.js [hunk 5]
```diff
@@ -306,7 +328,145 @@ module.exports = class bleutrade extends bittrex {
...
+            'before': undefined,
+            'after': undefined,
+            'status': 'ok',
+            'fee': fee,
+        };
+    }
+
+    async fetchLedger (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.v3PrivateGetGetmytransactions (params);
```
Comment (kroitor) [inline js/bleutrade.js]: @npomfret a quick question, the docs say that this endpoint requires an asset id in the params, but i can't see it in the code – does it really work without specifying the asset id ?https://app.swaggerhub.com/apis-docs/bleu/white-label/3.0.0#/Private%20functions/getMyTransactions

**Pull request merged**